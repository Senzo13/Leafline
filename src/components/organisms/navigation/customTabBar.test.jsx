import React from "react";
import renderer from "react-test-renderer";
import CustomTabBar from "./CustomTabBar";

describe("CustomTabBar", () => {
  it("renders correctly", () => {
    const state = {
      routes: [
        { key: "home", name: "Home" },
        { key: "dashboard", name: "Dashboard" },
        { key: "settings", name: "Settings" },
      ],
      index: 0,
    };

    const descriptors = {
      home: { options: { activeIcon: "home", inactiveIcon: "home-outline" } },
      dashboard: {
        options: { activeIcon: "dashboard", inactiveIcon: "dashboard-outline" },
      },
      settings: {
        options: { activeIcon: "settings", inactiveIcon: "settings-outline" },
      },
    };

    const navigation = {
      emit: jest.fn(),
      navigate: jest.fn(),
    };

    const tree = renderer
      .create(
        <CustomTabBar
          state={state}
          descriptors={descriptors}
          navigation={navigation}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
