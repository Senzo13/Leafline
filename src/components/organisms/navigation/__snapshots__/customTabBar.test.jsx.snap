// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTabBar renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "borderRadius": 20,
      "bottom": 24,
      "elevation": 8,
      "flexDirection": "row",
      "height": 60,
      "left": 16,
      "paddingBottom": 5,
      "paddingTop": 10,
      "position": "absolute",
      "right": 16,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#50B26C",
          "borderRadius": 20,
          "height": 3,
          "position": "absolute",
          "top": -10,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  />
</View>
`;
